File: core/src/main/java/com/github/jackchen/android/core/initializer/AndroidSampleInitializer.java
package com.github.jackchen.android.core.initializer;

import android.content.Context;
import androidx.annotation.NonNull;
import androidx.startup.Initializer;
import com.github.jackchen.android.core.AndroidSample;
import java.util.Collections;
import java.util.List;

public class AndroidSampleInitializer implements Initializer<AndroidSample> {
    @NonNull @Override public AndroidSample create(@NonNull final Context context) {
        final AndroidSample androidSample = AndroidSample.Companion.getInstance();
        androidSample.attachToContext(context);
        return androidSample;
    }

    @NonNull @Override public List<Class<? extends Initializer<?>>> dependencies() {
        return Collections.emptyList();
    }
}

File: core/src/main/java/com/github/jackchen/android/core/initializer/AndroidSampleInitializer.kt
package com.github.jackchen.android.core.initializer

import android.content.Context
import androidx.startup.Initializer
import com.github.jackchen.android.core.AndroidSample
import java.util.*

/**
 * @author JackChen & Airsaid
 */
class AndroidSampleInitializer : Initializer<AndroidSample> {
  override fun create(context: Context): AndroidSample {
    return AndroidSample.instance.apply {
      attachToContext(context)
    }
  }

  override fun dependencies(): MutableList<Class<out Initializer<*>>> {
    return Collections.emptyList()
  }
}
